### DISCUSSION TITLE
creating-a-validator-to-check-for-common-passwords-in-asp-net-core-identity/

### DISCUSSION THREAD

**Discussion by andrewlock** [andrewlock/blog-comments#221] [Announcements]
# creating-a-validator-to-check-for-common-passwords-in-asp-net-core-identity/

In this post, I introduce a package for ASP.NET Core Identity that lets you validate a user hasn't used one of the most common 100,000 passwords.

https://andrewlock.net/creating-a-validator-to-check-for-common-passwords-in-asp-net-core-identity/

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at July 19 2017, 08:47, in reply to [Muhammad Rehan Saeed](https://disqus.com/by/RehanSaeedUK/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/221#discussioncomment-6042723)</em>

---

<p>Yep, agreed - I want to use the IdentityOptions object to get that length though, as I don't want the possibility of mismatch between the common password and the overall length requirement. Pretty sure I looked at that though, and I don't think you can use DI with the validators... Need to double check and have a think!</p>

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at July 19 2017, 07:06, in reply to [Muhammad Rehan Saeed](https://disqus.com/by/RehanSaeedUK/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/221#discussioncomment-6042723)</em>

---

<p>Just pushed an update (v1.1.0) to the package that does this - filters out any passwords shorter than the minimum :)</p>

[a comment by an automated account is not shown]

**Reply by andrewlock** [reply]
<em>[Sock](https://disqus.com/by/andrewlockdotnet/) (@andrewlock) commented [on Disqus](https://disqus.com/home/forum/andrewlock/) at July 19 2017, 08:53, in reply to [Muhammad Rehan Saeed](https://disqus.com/by/RehanSaeedUK/)'s [comment](https://github.com/andrewlock/blog-comments/discussions/221#discussioncomment-6042727)</em>

---

<p>To be honest, I used that list for two reasons - it was easiest, and it's what they used in Discourse. The work of counting the top "X" passwords had already been done and split out nicely so it was really the simplest case. If you have a password list that's more representative of users of your app, then it should be easy enough to swap it out.</p><p>In terms of updating the list - I'll just keep an eye out for any big dumps and see how they pan out. I'm actually not sure how much the common passwords have changed over time - it's an interesting question.</p><p>As an aside, have you looked at the passwords in those lists? They're sooo bad it's unbelievable!</p>

[a comment by an automated account is not shown]

**Comment by carnahanliam** [comment]
I noticed you have a similar nuget package which validates using HaveIBeenPwned API. HaveIBeenPwned say they validate against hundreds of millions of passwords from data breaches, and this package is based on a list 10 million. Can you explain what inspired you to create these two separate packages? Am I safe to assume that using your PwnedPasswords.Validator alone should be secure enough since it uses a larger set of passwords, or would you use both?
